src/wymeditor/editor/dialogs.js

```diff
@@ -0,0 +1,501 @@
+    if (
+        dialog &&
+        dialog.shouldOpen.call(wym) !== true
+    ) {
+        return false;
+    }
+
+    if (pDialogWindowFeatures) {
+        // Provided argument
+        dialogWindowFeatures = pDialogWindowFeatures;
+    } else if (dialog && dialog.getWindowFeatures) {
+        // Dialog's code
+        dialogWindowFeatures = dialog.windowFeatures.call(wym);
+    } else if (options.dialogFeatures) {
+        // Provided option
+        dialogWindowFeatures = options.dialogFeatures;
+    } else {
+        // Default
+        dialogWindowFeatures = [
+            "menubar=no",
+            "titlebar=no",
+            "toolbar=no",
+            "resizable=no",
+            "width=560",
+            "height=300",
+            "top=0",
+            "left=0"
+        ].join(",");
+    }
+
+    wDialog = window.open(
+        '',
+        "wymDialogWindow",
+        dialogWindowFeatures
+    );
+
+    if (
+        typeof wDialog !== "object" ||
+        wDialog.window !== wDialog
+    ) {
+        WYMeditor.console.warn("Could not create a dialog window");
+        return false;
+    }
+
+    // In the case where a dialog window already exists, it will be reused.
+    // If it is in the background, behind another window, without the following
+    // `focus` call, it will remain in the background and the user may not
+    // understand where it is. This `focus` call brings the dialog to the
+    // foreground, making reasonably sure the user notices it.
+    wDialog.focus();
+
+    // Construct the dialog
+    dialogHtml = options.dialogHtml || String() +
+        '<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" ' +
+                '"http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">' +
+        '<html dir="' + WYMeditor.DIRECTION + '">' +
+            '<head>' +
+                '<title>' + WYMeditor.DIALOG_TITLE + '</title>' +
+            '</head>' +
+            WYMeditor.DIALOG_BODY +
+        '</html>';
+
+    // HTML template replacements
+    htmlStrReplacements = [
+        {
+            placeholder: WYMeditor.DIRECTION,
+            replacement: options.direction
+        },
+        {
+            placeholder: WYMeditor.DIALOG_TITLE,
+            replacement: dialog ? wym._encloseString(dialog.title) : "Dialog"
+        },
+        {
+            placeholder: WYMeditor.DIALOG_BODY,
+            replacement: pBodyHtml || dialog.getBodyHtml.call(wym)
+        }
+    ];
+
+    // Perform HTML template replacements
+    for (i = 0; i < htmlStrReplacements.length; i++) {
+        dialogHtml = WYMeditor.Helper.replaceAllInStr(
+            dialogHtml,
+            htmlStrReplacements[i].placeholder,
+            htmlStrReplacements[i].replacement
+        );
+    }
+
+    dialogHtml = wym.replaceStrings(dialogHtml);
+
+    doc = wDialog.document;
+    doc.write(dialogHtml);
+    if (dialog && dialog.getBodyClass) {
+        jQuery(doc.body).addClass(dialog.getBodyClass.call(wym));
+    }
+    doc.close();
```

Comment:
It [closes the document](https://developer.mozilla.org/en-US/docs/Web/API/document.close) after writing. Not sure whether it is required but it seems more explicit.